Extensions/P2P/B_p2ptools.js

```diff
@@ -0,0 +1,308 @@
+ * @param {string} [eventData] - Additional data to send with the event.
+ */
+gdjs.evtTools.p2p.sendDataToAll = function(eventName, eventData) {
+  for(var id in gdjs.evtTools.p2p.connections) {
+    gdjs.evtTools.p2p.connections[id].send({eventName: eventName, data: eventData});
+  }
+}
+
+/**
+ * Send an event to one specific connected client.
+ * @param {string} id - The id of the client to send the event to.
+ * @param {string} eventName - The event to trigger.
+ * @param {gdjs.Variable} variable - Additional variable to send with the event.
+ */
+gdjs.evtTools.p2p.sendVariableTo = function(id, eventName, variable) {
+  if(gdjs.evtTools.p2p.connections[id])
+    gdjs.evtTools.p2p.connections[id].send({
+      eventName: eventName, 
+      data: gdjs.evtTools.network.variableStructureToJSON(variable)
+    });
+}
+
+/**
+ * Send an event to all connected clients.
+ * @param {string} eventName - The event to trigger.
+ * @param {gdjs.Variable} variable - Additional variable to send with the event.
+ */
+gdjs.evtTools.p2p.sendVariableToAll = function(eventName, variable) {
+  for(var id in gdjs.evtTools.p2p.connections) {
+    gdjs.evtTools.p2p.connections[id].send({
+      eventName: eventName, 
+      data: gdjs.evtTools.network.variableStructureToJSON(variable)
+    });
+  }
+}
+
+/**
+ * Get some data associated to the last trigger of an event.
+ * @param {string} eventName - The event to get data from.
+ * @returns {string} - The data as JSON.
+ */
+gdjs.evtTools.p2p.getEventData = function(eventName) {
+  var dataLoss = gdjs.evtTools.p2p.eventHandling[eventName];
+  if (typeof dataLoss === "undefined" || dataLoss === false) {
+    var event = gdjs.evtTools.p2p.lastEventData[eventName];
+    return event[event.length - 1];
+  } else {
+    return gdjs.evtTools.p2p.lastEventData[eventName];
+  }
+}
+
+/**
+ * Get a variable associated to the last trigger of an event.
+ * @param {string} eventName - The event to get the variable from.
+ * @param {gdjs.Variable} variable - The variable where to store the variable content.
+ */
+gdjs.evtTools.p2p.getEventVariable = function(eventName, variable) {
+  gdjs.evtTools.network.jsonToVariableStructure(gdjs.evtTools.p2p.getEventData(eventName), variable);
+}
+
+/**
+ * Connects to a custom broker server.
+ * @param {string} host The host of the broker server
+ * @param {number} port
+ * @param {string} path The path (part of the url after the host) to the broker server
+ * @param {string} key Optional password to connect to the broker server
+ * @param {boolean} ssl Use ssl?
+ */
+gdjs.evtTools.p2p.useCustomBrokerServer = function(host, port, path, key, ssl) {
+  key = key.length === 0 ? "peerjs" : key; // All servers have "peerjs" as default key
+  gdjs.evtTools.p2p.peerConfig = {
+    debug: 1,
+    host,
+    port,
+    path,
+    secure: ssl,
+    key,
+  };
+  gdjs.evtTools.p2p.loadPeerJS();
+}
+
+/**
+ * Use default broker server. NOT RECOMMENDED.
```

Comment:
Use default broker server. This is not recommended for published games, this server should only be used for quick testing in development.

Reply:
I would also not recommend it in development as it can be down and cause confusion on why it isn't working.